lib/error.js

```diff
@@ -1,44 +1,65 @@
 "use strict";
 
+var util = require('util');
+
 /**
  * Creates a new MongoError
  * @class
  * @augments Error
- * @param {string} message The error message
+ * @param {Error|string|object} message The error message
+ * @property {string} message The error message
+ * @property {string} stack The error call stack
  * @return {MongoError} A MongoError instance
  */
 function MongoError(message) {
   this.name = 'MongoError';
-  this.message = message;
-  Error.captureStackTrace(this, MongoError);
+
+  if (message instanceof Error) {
+    this.message = message.message;
+    this.stack = message.stack;
+  } else {
+    if (typeof message === 'string') {
+      this.message = message;
+    } else {
+      this.message = message.message || message.errmsg || message.$err || 'n/a';
+      for (var name in message) {
+        this[name] = message[name];
+      }
+    }
+    Error.captureStackTrace(this, MongoError);
```

Comment:
also I think we _always_ want to capture the stack trace. If an error is passed in, you should include that as like an originating error, it shouldn't _become_ this error